artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -701,7 +745,11 @@ private void connectSender(Queue queue,
             sender.setSenderSettleMode(SenderSettleMode.UNSETTLED);
             sender.setReceiverSettleMode(ReceiverSettleMode.FIRST);
             Target target = new Target();
-            target.setAddress(targetName);
+            if (targetAddress != null && !targetAddress.isBlank()) {
+               target.setAddress(targetAddress);
```

Comment:
This targetName / targetAddress distinction is a bit awkard with both around given that 'target address' is usually the normal terminology and what the value actually sets. Perhaps change the prior one to targetAddress instead of targetName, then use targetAddressOverride for the new 'use this instead' value to clarify things?

Similarly for the receiver bit above.

EDIT: or see other comment, simplify this and move decision to the call site where config is being pulled originally.